### DISCUSSION TITLE
Resend queued messages

### DISCUSSION THREAD

**Discussion by gbhoffman** [nextgenhealthcare/connect#6240] [Q&A] [answered]
Using MirthConnect 4.20 on Windows Server.

 Recently, destination changed SFTP port numbers but word didn't get to me. 57 messages went into queue. Once I re-established communication, I manually re-processed the messages that were queued.

Is there a bulk way of re-sending queued messages once the channel opened up again?

Thanks,
Gary

**Comment by jonbartels** [comment, marked as answer]
Review this thread in Slack https://mirthconnect.slack.com/archives/C02SW0K4D/p1718897181586319

The users problem was algos but its the same concept for your change for ports. 

Todd Horst posted a highlighted solution: 
![image](https://github.com/nextgenhealthcare/connect/assets/7669292/c0e55c60-0c2d-4a71-acb0-5c341ab2bfc3)

You should read about "Regenerate Template" at https://docs.nextgen.com/ to make sure you understand all the implications.

**Reply by tonygermano** [reply]
"Regenerate Template" is required to pick up the port change. It will retain the same encodedData from the original message and pick up any changes to the destination connector itself. It will reapply velocity replacements, so if any global values or configuration map values have changed since the original run they will be picked up.

"Include Filter/Transformer" is not required as that will run the entire transformer again starting from the rawData, which may or may not produce the same encodedData and channelMap/connectorMap values. It could also potentially filter a message which passed the first time through, resulting in the message not being sent at all.
